### DISCUSSION TITLE
Build for Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by Anyeos** [olive-editor/olive#2254] [Show and tell]
Here the steps on how to build under Ubuntu 20.04 because it was a headache to solve conflicts on libraries (old versions that Ubuntu 20.04 carry and so on).

1. Uninstall the dev versions of packages (not the packages only the -dev): OpenEXR, OpenColorIO and OpenImageIO.
2. Build in your preferred ways: OpenEXR, OpenColorIO and OpenImageIO (in that order). Please take in account the needed versions and use a different path than your system libraries. I built: openexr-3.1, ocio-2.1 and oiio-2.3 (using the script on blender source code) all installed in /opt/lib.
3. Install qt5.15 with a ppa. I used "opt-qt-5.15.0-focal".
4. If you can, try OpenTimelineIO. It will compile but the linker will fail later when you try to build Olive. That is because OpenTimelineIO uses own Imath library and because it uses nonstd that I don't know how to solve. For now I not use OpenTimelineIO.
5. Clone olive master.
6. cd, and mkdir "build", and cd "build".
7. Create on the directory "build" and modify my script to point to your recently built libraries.
8. Execute the script and make.

If you have doubts please follow the Olive guide on how to build. I supposed you already did that.

Next, the script content. Create a file with the name you prefer and make it executable, fill it with the next content:
```
source /opt/qt515/bin/qt515-env.sh

OPENEXR_BASE_DIR=/opt/lib/openexr
OCIO_BASE_DIR=/opt/lib/ocio
OIIO_BASE_DIR=/opt/lib/oiio
OTIO_BASE_DIR=/usr/local


export PATH=$OPENEXR_BASE_DIR/bin:$OCIO_BASE_DIR/bin:$OIIO_BASE_DIR/bin:$PATH
export LD_LIBRARY_PATH=$OPENEXR_BASE_DIR/lib:$OCIO_BASE_DIR/lib:$OIIO_BASE_DIR/lib:$LD_LIBRARY_PATH
export PKG_CONFIG_PATH=$OPENEXR_BASE_DIR/lib/pkgconfig:$OCIO_BASE_DIR/lib/pkgconfig:$OIIO_BASE_DIR/lib/pkgconfig:$PKG_CONFIG_PATH


cmake .. \
-DCMAKE_BUILD_TYPE=RelWithDebInfo \
-DOCIO_BASE_DIR=$OCIO_BASE_DIR \
-DOCIO_LIBRARY=$OCIO_BASE_DIR/lib/libOpenColorIO.so -DOCIO_INCLUDE_DIR=$OCIO_BASE_DIR/include \
-DOIIO_BASE_DIR=$OIIO_BASE_DIR
```

### Update:
You must create a bash script replacing the olive-editor executable (and don't forget to rename the binary to something else like: olive-editor.bin). The script can be like this:
```
#!/bin/bash

source /opt/qt515/bin/qt515-env.sh
olive-editor.bin $*
```
It is because we are using the "opt-qt-5.15.0-focal" package, and if you don't do above it will try to use the system ones and will fail to run.

**Comment by AngeloFrancabandiera** [comment]
Well done! While I still think that 20.04 is the best Ubuntu ever, I have switched to the newer version because building Olive is way easier.

**Comment by Anyeos** [comment]
Yes it took me too much time but I prefer it over switching because I use NVIDIA and AI and the provided pre-compiled binaries are for Ubuntu 20.04. That may work with Ubuntu 22 but I prefer to stay with maximum compatibility and because it is working very well I prefer still not to switch.

Hope my guide can help somebody.

**Comment by motoseis** [comment]
I can't compile Olive on Ubuntu 22.04LTS.  Guess I'll try using your solution, or wait until 24.04 is ready for installation.  Thanks for posting your solution!

**Comment by alexmyczko** [comment]
I would like to get the Debian sid package into shape. If anyone wants to help...
